main.tf

```diff
@@ -0,0 +1,666 @@
+          "ec2:DescribeManagedPrefixLists",
+          "ec2:DescribeSecurityGroups",
+          "ec2:DescribeSecurityGroupRules",
+          "ec2:RevokeSecurityGroupEgress",
+          "ec2:AuthorizeSecurityGroupEgress",
+          "ec2:CreateTags",
+        ]
+        Resource = "*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "ec2:DeleteSecurityGroup",
+        ]
+        Resource = "*"
+        Condition = {
+          StringEquals = {
+            "aws:ResourceTag/Owner" = "CG.AWP"
+          }
+        }
+      },
+    ]
+  })
+}
+
+# The CloudGuardAWPCrossAccountRolePolicy_SaaS resource defines an IAM policy that is used to define the permissions for the CloudGuardAWPCrossAccountRole in SaaS mode.
+resource "aws_iam_policy" "CloudGuardAWPCrossAccountRolePolicy_SaaS" {
+  count       = local.scan_mode == "saas" ? 1 : 0
+  name        = "CloudGuardAWPCrossAccountRolePolicy_SaaS"
+  description = "Policy for CloudGuard AWP Cross Account Role - SaaS Mode"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "kms:DescribeKey",
+          "kms:ReplicateKey",
+        ]
+        Resource = [aws_kms_key.CloudGuardAWPKey[count.index].arn]
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "kms:PutKeyPolicy",
+          "kms:ScheduleKeyDeletion",
+          "kms:CancelKeyDeletion",
+          "kms:TagResource",
+        ]
+        Resource = aws_kms_key.CloudGuardAWPKey[count.index].arn
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "kms:CreateKey",
+        ]
+        Resource = "*"
+      },
+    ]
+  })
+}
+
+# The CloudGuardAWPCrossAccountRolePolicyAttachment resource attaches the CloudGuardAWPCrossAccountRolePolicy to the CloudGuardAWPCrossAccountRole.
+resource "aws_iam_policy_attachment" "CloudGuardAWPCrossAccountRolePolicyAttachment" {
+  count      = local.scan_mode == "inAccount" ? 1 : 0
+  name       = "CloudGuardAWPCrossAccountRolePolicyAttachment"
+  policy_arn = aws_iam_policy.CloudGuardAWPCrossAccountRolePolicy_InAccount[count.index].arn
+  roles      = [aws_iam_role.CloudGuardAWPCrossAccountRole.name]
+}
+
+# The CloudGuardAWPCrossAccountRolePolicyAttachment_SaaS resource attaches the CloudGuardAWPCrossAccountRolePolicy_SaaS to the CloudGuardAWPCrossAccountRole.
+resource "aws_iam_policy_attachment" "CloudGuardAWPCrossAccountRolePolicyAttachment_SaaS" {
+  count      = local.scan_mode == "saas" ? 1 : 0
+  name       = "CloudGuardAWPCrossAccountRolePolicyAttachment_SaaS"
+  policy_arn = aws_iam_policy.CloudGuardAWPCrossAccountRolePolicy_SaaS[count.index].arn
+  roles      = [aws_iam_role.CloudGuardAWPCrossAccountRole.name]
+}
+# END Cross account role policy
+
+# The CloudGuardAWPSnapshotsUtilsFunctionZip resource defines http data source to download the remote function file from S3 pre-signed URL.
+data "http" "CloudGuardAWPSnapshotsUtilsFunctionZip" {
```

Comment:
this resouce have the name CloudGuardAWPSnapshotsUtilsFunctionZip and the next resource have the same name, maybe change them to be unique names give fast differentiation for them